lib/Doctrine/DBAL/Connection.php

```diff
@@ -929,16 +926,14 @@ public function executeQuery($query, array $params = [], $types = [], QueryCache
     /**
      * Executes a caching query.
      *
-     * @param string                                 $query  The SQL query to execute.
-     * @param array                                  $params The parameters to bind to the query, if any.
-     * @param array                                  $types  The types the previous parameters are in.
-     * @param \Doctrine\DBAL\Cache\QueryCacheProfile $qcp    The query cache profile.
-     *
-     * @return \Doctrine\DBAL\Driver\ResultStatement
+     * @param string              $query  The SQL query to execute.
+     * @param mixed[]             $params The parameters to bind to the query, if any.
+     * @param (int|string)|Type[] $types  The types the previous parameters are in.
+     * @param QueryCacheProfile   $qcp    The query cache profile.
      *
-     * @throws \Doctrine\DBAL\Cache\CacheException
+     * @throws DBALException
```

Comment:
`CacheException` is a subclass of `DBALException` which can happen underneath. Therefore, replaced.